test/e2e/set/scyllacluster/scyllacluster_tls.go

```diff
@@ -244,6 +250,154 @@ var _ = g.Describe("ScyllaCluster", func() {
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+		defer cancel()
+
+		sc := scyllafixture.BasicScyllaCluster.ReadOrFail()
+		o.Expect(sc.Spec.Datacenter.Racks).To(o.HaveLen(1))
+		sc.Spec.Datacenter.Racks[0].Members = 1
+
+		framework.By("Creating a ScyllaCluster")
+		sc, err := f.ScyllaClient().ScyllaV1().ScyllaClusters(f.Namespace()).Create(ctx, sc, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for the ScyllaCluster to rollout (RV=%s)", sc.ResourceVersion)
+		waitCtx1, waitCtx1Cancel := utils.ContextForRollout(ctx, sc)
+		defer waitCtx1Cancel()
+		sc, err = utils.WaitForScyllaClusterState(waitCtx1, f.ScyllaClient().ScyllaV1(), sc.Namespace, sc.Name, utils.WaitForStateOptions{}, utils.IsScyllaClusterRolledOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		verifyScyllaCluster(ctx, f.KubeClient(), sc)
+		hosts := getScyllaHostsAndWaitForFullQuorum(ctx, f.KubeClient().CoreV1(), sc)
+		o.Expect(hosts).To(o.HaveLen(1))
+		host := hosts[0]
+
+		servingCABundleConfigMap, err := f.KubeClient().CoreV1().ConfigMaps(f.Namespace()).Get(ctx, fmt.Sprintf("%s-local-serving-ca", sc.Name), metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		servingCACerts, _ := verification.VerifyAndParseCABundle(servingCABundleConfigMap)
+		o.Expect(servingCACerts).To(o.HaveLen(1))
+
+		adminClientSecret, err := f.KubeClient().CoreV1().Secrets(f.Namespace()).Get(ctx, fmt.Sprintf("%s-local-user-admin", sc.Name), metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		_, adminClientCertBytes, _, adminClientKeyBytes := verification.VerifyAndParseTLSCert(adminClientSecret, verification.TLSCertOptions{
+			IsCA:     pointer.Ptr(false),
+			KeyUsage: pointer.Ptr(x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature),
+		})
+
+		servingCAPool := x509.NewCertPool()
+		servingCAPool.AddCert(servingCACerts[0])
+		adminTLSCert, err := tls.X509KeyPair(adminClientCertBytes, adminClientKeyBytes)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		var wg sync.WaitGroup
+		defer wg.Wait()
+
+		trafficCtx, trafficCtxCancel := context.WithCancel(ctx)
+		defer trafficCtxCancel()
+
+		const workers = 3
+
+		for i := 0; i < workers; i++ {
+			trafficStarted := make(chan struct{})
+			notifyTrafficStarted := sync.OnceFunc(func() {
+				close(trafficStarted)
+			})
+
+			wg.Add(1)
+			go func(idx int) {
+				defer wg.Done()
+				defer g.GinkgoRecover()
+
+				clusterConfig := gocql.NewCluster(hosts...)
+				clusterConfig.Port = 9142
+				clusterConfig.SslOpts = &gocql.SslOptions{
+					Config: &tls.Config{
+						ServerName:         host,
+						InsecureSkipVerify: false,
+						Certificates:       []tls.Certificate{adminTLSCert},
+						RootCAs:            servingCAPool,
+					},
+				}
+				session, err := gocqlx.WrapSession(clusterConfig.CreateSession())
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				di := insertAndVerifyCQLData(ctx, hosts, utils.WithSession(&session))
```

Comment:
Also, maybe instead of reading in the loop here, we could have a method that reads continuously in the background and takes context/stop channel as parameter? Seems like something useful in the longer run

Reply:
done